src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -775,8 +787,10 @@ static void create_rr_locked(grpc_exec_ctx *exec_ctx, glb_lb_policy *glb_policy,
 /* glb_policy->rr_policy may be NULL (initial handover) */
 static void rr_handover_locked(grpc_exec_ctx *exec_ctx,
                                glb_lb_policy *glb_policy) {
-  GPR_ASSERT(glb_policy->serverlist != NULL &&
-             glb_policy->serverlist->num_servers > 0);
+  GPR_ASSERT((glb_policy->fallback && glb_policy->fallback_addresses != NULL &&
```

Comment:
At the very least, we can separate the OR-ed clauses:

```
GPR_ASSERT(glb_policy->fallback &c& glb_policy->fallback_addresses != NULL && glb_policy->fallback_addresses->num_addresses > 0);
GPR_ASSERT(glb_policy->serverlist != NULL && glb_policy->serverlist->num_servers > 0);
```
Moreover, it you take Mark's suggestion to get rid of the boolean in favor of using `glb_policy->fallback_addresses != NULL` as the check, the first assert is further simplified and looks very much like the second.